assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -117,7 +128,22 @@ function useUploader() {
         ..._additionalData,
       };
 
-      if (canTranscodeFile(file)) {
+      // Transcoding is enabled, let's give it a try!
+      if (isTranscodingEnabled(file)) {
+        // Bail early if the file is too large for transcoding.
+        if (isFileTooLarge(file)) {
+          const message = sprintf(
+            /* translators: %s: File size. */
+            __(
+              'Your file is too large (%s MB) and cannot be processed. Please try again with a smaller file.',
```

Comment:
Could we add the max file size in here similar to approve? 
```suggestion
              ''Your file is too large at %1$sMB and the upload limit is %2$sMB. Please try again with a smaller file.',
```

Reply:
It would need different wording, but theoretically, yes. I'll look into it.